### PR TITLE
Remove str concatenation

### PR DESCRIPTION
**Pull request by justmobilize**

### CODE REVIEW

File: adafruit_requests.py [hunk 1]
```diff
@@ -369,22 +369,22 @@ def _build_boundary_data(self, files: dict):  # pylint: disable=too-many-locals
             file_name = field_values[0]
             file_handle = field_values[1]
 
-            boundary_data = f"--{boundary_string}\r\n"
-            boundary_data += f'Content-Disposition: form-data; name="{field_name}"'
+            boundary_objects.append(
```
Comment (justmobilize) [inline adafruit_requests.py:372]: add all strings to the list

File: adafruit_requests.py [hunk 2]
```diff
@@ -400,19 +400,15 @@ def _build_boundary_data(self, files: dict):  # pylint: disable=too-many-locals
...
+            boundary_objects.append("\r\n")
 
-        boundary_data = f"--{boundary_string}--\r\n"
+        boundary_objects.append(f"--{boundary_string}--\r\n")
 
-        content_length += len(boundary_data)
-        boundary_objects.append(boundary_data)
+        for boundary_object in boundary_objects:
+            if isinstance(boundary_object, str):
+                content_length += len(boundary_object)
```
Comment (justmobilize) [inline adafruit_requests.py:411]: and calculate the length at the end

File: tox.ini [hunk 3]
```diff
@@ -10,13 +10,15 @@ envlist = py311
 description = run tests
 deps =
     pytest==7.4.3
+    requests
```
Comment (justmobilize) [inline tox.ini:13]: needed to run tox, since requests is now in optional

**Comment by FoamyGuy** [comment]
Thank you! 

Looks good to me, I tested this successfully on a Feather S3 TFT.

**Pull request merged**